WordPress/Classes/Utility/Media/MediaImageExporter.swift

```diff
@@ -59,6 +55,42 @@ class MediaImageExporter: MediaExporter {
     ///
     fileprivate let defaultImageFilename = "image"
 
+    private let image: UIImage?
+    private let data: Data?
+    private let url: URL?
+    private let filename: String?
+
+    init(image: UIImage, filename: String?) {
```

Comment:
I think these could all delegate to another initialiser that takes all the values as parameters – would reduce each of these to one line?

Reply:
Good suggestion, going to implement that.

Reply:
Actually this is not a good idea, for example if you set an UIImage, you should set a Data, the same applies to the URL.

Reply:
I'm not sure I follow? I was thinking something like this:

```
private init(image: UIImage?, filename: String?, data: Data?, url: URL?) {
    self.image = image
    self.filename = filename
    self.data = data
    self.url = url
}

convenience init(image: UIImage, filename: String?) {
    self.init(image: image, filename: filename, data: nil, url: nil)
}

convenience init(url: URL) {
    self.init(image: nil, filename: nil, data: nil, url: url)
}

convenience init(data: Data, filename: String?) {
    self.init(image: nil, filename: filename, data: data, url: nil)
}
```